logstash-core/lib/logstash/api/commands/default_metadata.rb

```diff
@@ -20,7 +20,9 @@ def version
         end
 
         def http_address
-          service.agent.webserver.address
+          @http_address ||= service.get_shallow(:http_address).value
+        rescue ::LogStash::Instrument::MetricStore::MetricNotFound, NoMethodError => e
+          nil
```

Comment:
NoMethod error in which case we can get that, if a metric is returned the #value should exist.